flask_annex/file.py

```diff
@@ -57,12 +57,16 @@ def get_file(self, key, out_file):
                 shutil.copyfileobj(in_fp, out_file)
 
     def list_keys(self, prefix):
-        pattern = self._get_filename('{}*'.format(prefix))
-        filenames = glob.glob(pattern)
+        prefix = os.path.normpath(prefix)
+        dir_pattern = self._get_filename('{}/**/*'.format(prefix))
```

Comment:
I think I tried without getting the expected results, I'll try again, and add tests